src/Symfony/Bundle/FrameworkBundle/Debug/TraceableEventDispatcher.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpKernel\Debug\DebugRequestStack;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher as BaseTraceableEventDispatcher;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * @internal
+ */
+final class TraceableEventDispatcher extends BaseTraceableEventDispatcher
+{
+    public function __construct(EventDispatcherInterface $dispatcher, Stopwatch $stopwatch, LoggerInterface $logger = null, private ?DebugRequestStack $requestStack = null)
+    {
+        parent::__construct($dispatcher, $stopwatch, $logger, $requestStack);
+    }
+
+    protected function beforeDispatch(string $eventName, object $event): void
+    {
+        if (!class_exists(ConsoleEvents::class)) {
+            parent::beforeDispatch($eventName, $event);
+
+            return;
+        }
+
+        switch ($eventName) {
+            case ConsoleEvents::COMMAND:
+                $request = $this->requestStack->getCurrentRequest(ignoreVirtual: false);
+
+                /** @var ConsoleCommandEvent $event */
+                if (!$request instanceof CliRequest || $request->command !== $event->getCommand()) {
+                    break;
+                }
+
+                $request->attributes->set('_stopwatch_token', substr(hash('sha256', uniqid(mt_rand(), true)), 0, 6));
```

Comment:
Do we need a special child class of TraceableEventDispatcher for that ? Couldn't we do that in a listener on the `ConsoleEvents::COMMAND` that would be registered with a very high priority instead ?

Reply:
My very first implementation was made that way, but a listener holding that logic is not collected.
Also what priority would be high enough to not miss earlier listeners?

Reply:
what do you mean by "it is not traced" ?

Reply:
It is not shown is the performance profiler panel, not a big deal I agree, but it prevents knowing its impact on global performance.
Also anything before it will not be shown as well.